lua/supermaven-nvim/util.lua

```diff
@@ -153,8 +153,15 @@ function M.ends_with(str, suffix)
 end
 
 function M.get_utf8_length(str)
-  local byte_array = { str.byte(str, 1, -1) }
-  return #byte_array
+    local ALLOWED_LENGTH = 8000 -- maximum length .byte function may take in LuaJIT (used by Neovim)
+    local splitted_str = {}
+
+    for i = 1, #str, ALLOWED_LENGTH do
+        table.insert(splitted_str, string.sub(str, i, i + ALLOWED_LENGTH - 1))
+        i = i + ALLOWED_LENGTH
+    end
+
+    return #vim.iter(splitted_str):map(function(s) return { s.byte(s, 1, -1) } end):flatten():totable()
```

Comment:
I was getting an error "flatten on nil" was unable to test any further.

Reply:
That's interesting,
What does `nvim -v` output?
Could you provide a little file where this happens?

Reply:
```
NVIM v0.10.0-dev-1840+g38b0bb3c9-Homebrew
Build type: Release
LuaJIT 2.1.1713773202
```

It happened for me on basically ever action.

Reply:
<img width="935" alt="Screenshot 2024-05-12 at 08 07 31" src="https://github.com/supermaven-inc/supermaven-nvim/assets/866743/26d72c9f-1926-4713-8d3c-696517063095">

The lua lsp also seems to complain about it.

Reply:
That's super duper weird, flatten has been in the lua api since the beginning if I'm guessing correctly
If you don't mind dropping the output of `nvim -V1 -v`
for me it says:
```
NVIM v0.10.0-dev-3107+g8c7a8be27
Build type: RelWithDebInfo
LuaJIT 2.1.1713484068
Compilation: /usr/bin/gcc-10 -O2 -g -Og -g -flto=auto -fno-fat-lto-objects -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fsigned-char -fstack-protector-strong -Wno-conversion -fno-common -Wno-unused-result -Wimplicit-fallthrough -fdiagnostics-color=always  -DUNIT_TESTING -DHAVE_UNIBILIUM -D_GNU_SOURCE -DINCLUDE_GENERATED_DECLARATIONS -I/home/runner/work/neovim/neovim/.deps/usr/include/luajit-2.1 -I/home/runner/work/neovim/neovim/.deps/usr/include -I/home/runner/work/neovim/neovim/build/src/nvim/auto -I/home/runner/work/neovim/neovim/build/include -I/home/runner/work/neovim/neovim/build/cmake.config -I/home/runner/work/neovim/neovim/src -I/usr/include
```
If you could compile neovim from source, that'd be really helpful

Reply:
After rebuilding neovim it's available. 👍

Reply:
See https://github.com/supermaven-inc/supermaven-nvim/pull/7#issuecomment-2106436751